packages/core/src/analytics/hubspotClient.ts

```diff
@@ -0,0 +1,63 @@
+import { Client } from "@hubspot/api-client";
+import { ApiException } from "@hubspot/api-client/lib/codegen/crm/contacts/apis/exception";
+
+const accessToken = process.env.HUBSPOT_ACCESS_TOKEN;
+if (!accessToken) {
+  throw new Error("Missing HUBSPOT_ACCESS_TOKEN");
+}
+
+const hubspotClient = new Client({ accessToken });
+
+interface CreateHubspotCustomerInput {
+  email: string;
+  firstName: string | null;
+  lastName: string | null;
+  marketingAccepted: boolean;
+}
+
+export const createHubspotCustomer = async ({
+  email,
+  firstName,
+  lastName,
+  marketingAccepted,
+}: CreateHubspotCustomerInput) => {
+  let id: string | undefined;
+  try {
+    const result = await hubspotClient.crm.contacts.basicApi.getById(
+      email,
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      "email",
+    );
+    id = result.id;
+  } catch (e) {
+    if (e instanceof ApiException && e.code === 404) {
```

Comment:
What's the 404 saying here?

If it means no user found then we wouldn't want to return early, right?

Reply:
Guessing it doesn't mean that if the code is successfully creating new hubspot entities

Reply:
Oh, you're right. Well spotted as I thought I had tested that